datalog/datalog.go

```diff
@@ -44,56 +46,66 @@ func (v Set) Equal(t ID) bool {
 		cmap[v] = struct{}{}
 	}
 
-	for _, id := range v {
+	for _, id := range s {
 		if _, ok := cmap[id]; !ok {
 			return false
 		}
 	}
 	return true
 }
+func (s Set) String() string {
+	eltStr := make([]string, 0, len(s))
+	for _, e := range s {
+		eltStr = append(eltStr, e.String())
+	}
+
```

Comment:
I think this should be sorted, for consistency, as the map will randomize the order.